### PR TITLE
#756 autocomplete suggestions from backend

### PR DESCRIPTION
**Pull request by mspratt-biot**

### CODE REVIEW

File: udmif/api/src/device/resolvers.ts [hunk 1]
```diff
@@ -15,5 +15,10 @@ export const resolvers = {
     points: (_, { deviceId }, { dataSources: { deviceDS } }) => {
       return deviceDS.getPoints(deviceId);
     },
+    autocompleteSuggestions: (_, { autocompleteOptions }, { dataSources: { deviceDS } }) => {
```
[the remark this answers is not in the document]
Comment (neil-buildingsiot) [reply udmif/api/src/device/resolvers.ts:18]: I'm skeptical about calling this autocompleteSuggestions as it really ties the API to the UX. Is this going to be returning specific types of entities? Could this be some device search (ie, devices resolver with filtering options) instead?

**Review by neil-buildingsiot — APPROVED** [review]

File: udmif/api/src/device/resolvers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mspratt-biot) [reply udmif/api/src/device/resolvers.ts:18]: Yea, I wasn't sure what to call this one. I do need an API to fetch a slice of some unique results which match what the users typed in. autocompleteOptions is an object with fields `entity`,`field`,`term`, and `limit`. The intention is to keep this as general as possible, not tied to just devices.

[the remark this answers is not in the document]
Comment (neil-buildingsiot) [reply udmif/api/src/device/resolvers.ts:18]: Could we do something like have it where once they pick the entity type (ex: device), then you query the device api (using term for filtering/criteria). That way the API sticks to the domain without being tied to UX.

[the remark this answers is not in the document]
Comment (mspratt-biot) [reply udmif/api/src/device/resolvers.ts:18]: Hm, you know I didn't even think of doing it that way. This is a great idea. Let me see if I can make this work, then we can eliminate my other PR completely for the backend.

**Pull request closed without merging**